includes/class-wc-template-loader.php

```diff
@@ -193,7 +193,7 @@ public static function comments_template_loader( $template ) {
 	 * @since 3.3.0
 	 */
 	public static function unsupported_theme_init() {
-		if ( self::$shop_page_id ) {
+		if ( self::$shop_page_id && intval( self::$shop_page_id ) > 0 ) {
```

Comment:
`wc_get_page_id` always returns `int`. It uses absint or returns -1. Therefore it should be safe to check > 0 only.

This should also use yoda condition, So `0 < self::$shop_page_id`